services/cd-service/pkg/mapper/environments_config_test.go

```diff
@@ -525,6 +563,98 @@ func TestMapEnvironmentsToGroup(t *testing.T) {
 						makeEnv(nameProdFr, nameProd, makeUpstreamEnvironment(nameStagingFr), 2, api.Priority_PROD),
 					},
 					DistanceToUpstream: 2,
+					Priority:           api.Priority_PROD,
+				},
+			},
+		},
+		{
+			Name: "Environments with different environment priorities",
+			InputEnvs: map[string]config.EnvironmentConfig{
```

Comment:
As the test scenario is quite complex, please add a comment specifying the conditions of the test. It increases readability and makes the test easier to understand

Reply:
Fair enough. Added a diagram :D Is it good?